### ISSUE TITLE
Using PackAsToolShimRuntimeIdentifiers and netcoreapp3.0 fails with error NETSDK1065

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/sdk#2867]
**Repro**

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <PackAsTool>true</PackAsTool>
    <PackAsToolShimRuntimeIdentifiers>win-x64;win-x86</PackAsToolShimRuntimeIdentifiers>
  </PropertyGroup>

</Project>
```
`dotnet pack`

**Result**
```
C:\Users\namc\.dotnet\x64\sdk\3.0.100-preview-010166\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(228,5): error NETSDK1065: Cannot find app host for win-x64. win-x64 could be an invalid runtime identifier (RID). For more information about RID, see https://aka.ms/rid-catalog. [C:\src\aspnet\BuildTools\repro\CliTool\CliTool.csproj]
```

**Details**

Using 3.0.100-preview-010166

cc @nguerrera @wli3

**Comment by natemcmaster** [comment]
Marking as blocking-partner because we cannot update aspnet to build with a new SDK until this is resolved.

**Comment by wli3** [comment]
ack

**Comment by natemcmaster** [comment]
Not for preview 2, but it will be for preview 3

**Comment by bricelam** [comment]
Any update on this? We're trying to work on our C# 8 related features and could really use the latest version of the SDK...

**Comment by bricelam** [comment]
cc @ajcvickers

**Comment by peterhuene** [comment]
The fix has been merged into `dotnet/sdk`, so now the change just needs to make its way to `dotnet/core-sdk` for a SDK build to contain the fix.

**Comment by peterhuene** [comment]
There was a conflict in the toolset maestro PR which I've now resolved to help it along its way.

**Comment by bricelam** [comment]
Still seeing this in 3.0.100-preview-010234. Is it expected to be fixed in there?

**Comment by peterhuene** [comment]
Unfortunately it looks like `dotnet/core-setup` never took a newer toolset (6 days old still) and as a result the SDK does not yet contain the fix.

We're noticing build failures in multiple repos which I assume is arcade-related; perhaps it is preventing Maestro from updating the toolset dependency in `dotnet/core-setup`?

@livarcocc should we manually update the toolset to one that contains @wli3's fix to unblock repos with tool shims that would like to move to a newer 3.0 build?

**Comment by wli3** [comment]
let me babysit a build for now

**Comment by wli3** [comment]
https://github.com/dotnet/core-sdk/pull/451 manual update here

**Comment by wli3** [comment]
@bricelam @natemcmaster sorry for the long delay. I verified the latest core sdk build does not repro the bug.

**Comment by natemcmaster** [comment]
Thanks for confirming the fix @wli3

**Comment by bricelam** [comment]
Still seeing this on 3.0.100-preview3-010281

**Comment by wli3** [comment]
The one I verified is 3.0.100-preview4-010309. Are you downloading from https://github.com/dotnet/core-sdk ?

**Comment by bricelam** [comment]
So it won’t be fixed for preview 3? Ok, we should be able to ship preview 3 using our current version of the SDK

**Comment by nguerrera** [comment]
It might be codeflow backed up in preview 3. Toolset hasn't gone in for a while. https://github.com/dotnet/core-sdk/pull/506

**Comment by peterhuene** [comment]
This was fixed in master, but doesn't appear to be in `release/3.0.1xx`.

**Comment by vijayrkn** [comment]
@wli3 - Does the current preview3 version have this fix?
https://github.com/dotnet/core-sdk/blob/master/eng/Versions.props#L15 
3.0.100-preview3.19123.1

**Comment by vijayrkn** [comment]
Scaffolding build is blocked because of this. If you can confirm the version, I will send a PR to update the KoreBuild to this version.

**Comment by wli3** [comment]
I just checked 3.0.100-preview3-010373 is a good

**Comment by vijayrkn** [comment]
Thanks! I have updated the KoreBuild to use 3.0.100-preview4-010309. If we run into any other issues. I will lower it to the above preview3 version.
